src/main/java/com/mongodb/kafka/connect/sink/MongoSinkTopicConfig.java

```diff
@@ -278,32 +273,21 @@ public MongoNamespace getNamespace() {
     return namespace;
   }
 
-  @SuppressWarnings("unchecked")
+  private <T> T configureInstance(final T instance) {
+    if (instance instanceof Configurable) {
+      ((Configurable) instance).configure(this);
+    }
+    return instance;
+  }
+
   public IdStrategy getIdStrategy() {
     if (idStrategy == null) {
-      String className = getString(DOCUMENT_ID_STRATEGY_CONFIG);
-      Optional<FieldProjector> fieldProjector = getFieldProjector(className);
       idStrategy =
-          createInstance(
-              DOCUMENT_ID_STRATEGY_CONFIG,
-              className,
-              IdStrategy.class,
-              () -> {
-                Class<IdStrategy> clazz = (Class<IdStrategy>) Class.forName(className);
-                boolean hasFieldProjectorConstructor =
-                    fieldProjector.isPresent()
-                        && Arrays.stream(clazz.getConstructors())
-                            .anyMatch(
-                                i ->
-                                    i.getParameterTypes().length == 1
-                                        && i.getParameterTypes()[0].equals(FieldProjector.class));
-                if (hasFieldProjectorConstructor) {
-                  return clazz
-                      .getConstructor(FieldProjector.class)
-                      .newInstance(fieldProjector.get());
-                }
-                return clazz.getConstructor().newInstance();
-              });
+          configureInstance(
```

Comment:
The new `configureInstance` helper method really simplifies the creation of the IdStrategy and provides an extendable pattern for users going forward.

It could be backwards breaking if any users added their own `IdStrategy` and relied on this undocumented behaviour.